### ISSUE TITLE
Cannot start minishift again after stopping it because of loop back storage issue

### ISSUE THREAD

**Issue by javilinux** [minishift/minishift#1639]
I'm using minishift v1.7.0+1549135 on a Fedora 26 with qemu-kvm-2.9.1-1.fc26.x86_64

I can start minishift the first time

```
$ minishift start 
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present and active ... OK
-- Starting local OpenShift cluster using 'kvm' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM ..................... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 0% used OK
-- OpenShift cluster will be configured with ...
   Version: v3.6.0
-- Checking `oc` support for startup flags ... 
   host-pv-dir ... OK
   host-volumes-dir ... OK
   routing-suffix ... OK
   host-config-dir ... OK
   host-data-dir ... OK
Starting OpenShift using openshift/origin:v3.6.0 ...
Pulling image openshift/origin:v3.6.0
Pulled 1/4 layers, 26% complete
Pulled 2/4 layers, 54% complete
Pulled 3/4 layers, 82% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.37:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

```

- Then I stop it:

```
$ minishift stop
Stopping local OpenShift cluster...
Cluster stopped.

```
- And it fails when trying to start it again:

```
$ minishift start
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present and active ... OK
-- Starting local OpenShift cluster using 'kvm' hypervisor ...
-- Starting Minishift VM ............. OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 11% used OK
Error determining Docker settings: ssh command error:
command : docker version --format '{{.Server.APIVersion}}'
err     : exit status 1
output  : 
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

````
Issue happen with both with b2d and centos.

Output of _systemctl status docker_ when using centos image:

```
# systemctl status docker -l
* docker.service - Docker Application Container Engine
   Loaded: loaded (/etc/systemd/system/docker.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Mon 2017-10-30 18:38:06 EDT; 7min ago
     Docs: http://docs.docker.com
  Process: 773 ExecStart=/usr/bin/dockerd-current -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --selinux-enabled --log-driver=journald --signature-verification=false --add-runtime docker-runc=/usr/libexec/docker/docker-runc-current --default-runtime=docker-runc --exec-opt native.cgroupdriver=systemd --userland-proxy-path=/usr/libexec/docker/docker-proxy-current --storage-driver devicemapper --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem (code=exited, status=1/FAILURE)
  Process: 771 ExecStartPre=/usr/local/bin/minishift-cert-gen (code=exited, status=0/SUCCESS)
 Main PID: 773 (code=exited, status=1/FAILURE)

Oct 30 18:38:04 localhost systemd[1]: Starting Docker Application Container Engine...
Oct 30 18:38:05 localhost dockerd-current[773]: time="2017-10-30T18:38:05.320402921-04:00" level=info msg="libcontainerd: new containerd process, pid: 838"
Oct 30 18:38:06 localhost dockerd-current[773]: time="2017-10-30T18:38:06.497267889-04:00" level=warning msg="devmapper: Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man docker` to refer to dm.thinpooldev section."
Oct 30 18:38:06 localhost dockerd-current[773]: time="2017-10-30T18:38:06.524662017-04:00" level=warning msg="devmapper: Base device already exists and has filesystem xfs on it. User specified filesystem  will be ignored."
Oct 30 18:38:06 localhost dockerd-current[773]: time="2017-10-30T18:38:06.618670330-04:00" level=fatal msg="Error starting daemon: layer does not exist"
Oct 30 18:38:06 localhost systemd[1]: docker.service: main process exited, code=exited, status=1/FAILURE
Oct 30 18:38:06 localhost systemd[1]: Failed to start Docker Application Container Engine.
Oct 30 18:38:06 localhost systemd[1]: Unit docker.service entered failed state.
Oct 30 18:38:06 localhost systemd[1]: docker.service failed.
```
Also tested with minishift _v1.8.0+6455fc3_

**Comment by javilinux** [comment]
Forgot to coment that deleting the vm an starting it again solves the issue.

This is a relevant snippet when running with 

> --show-libmachine-logs --v=5 

```
'sudo docker version' output:

.checking docker daemon
(minishift) Calling .GetSSHHostname
(minishift) DBG | GetIP called for minishift
(minishift) DBG | Failed to retrieve dnsmasq leases from /var/lib/libvirt/dnsmasq/docker-machines.leases
(minishift) DBG | IP address: 192.168.42.36
(minishift) DBG | Unable to locate IP address for MAC 52:54:00:9c:7b:84
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) DBG | AK: resolvestorepath: /home/jaramire/.minishift
(minishift) Calling .GetSSHKeyPath
(minishift) DBG | AK: resolvestorepath: /home/jaramire/.minishift
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /home/jaramire/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.42.36 -o IdentitiesOnly=yes -i /home/jaramire/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo docker version
SSH cmd err, output: exit status 1: Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

Error getting SSH command to check if the daemon is running: ssh command error:
command : sudo docker version
err     : exit status 1
output  : Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

'sudo docker version' output:

.checking docker daemon
(minishift) Calling .GetSSHHostname
(minishift) DBG | GetIP called for minishift
(minishift) DBG | Failed to retrieve dnsmasq leases from /var/lib/libvirt/dnsmasq/docker-machines.leases
(minishift) DBG | IP address: 192.168.42.36
(minishift) DBG | Unable to locate IP address for MAC 52:54:00:9c:7b:84
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) DBG | AK: resolvestorepath: /home/jaramire/.minishift
(minishift) DBG | AK: resolvestorepath: /home/jaramire/.minishift
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /home/jaramire/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.42.36 -o IdentitiesOnly=yes -i /home/jaramire/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo docker version
SSH cmd err, output: exit status 1: Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

Error getting SSH command to check if the daemon is running: ssh command error:
command : sudo docker version
err     : exit status 1
output  : Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

'sudo docker version' output:

```

**Comment by rmarting** [comment]
I have not any issue with minishift in Fedora26 to start/stop life cycle.

My environment is:
* Minishift:
```
$ minishift version
minishift v1.8.0+6455fc3

```
* Docker:
```
$ docker version
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: <unknown>
 Go version:      go1.8.3
 Git commit:      b5e3294/1.13.1
 Built:           Wed Aug 16 13:58:20 2017
 OS/Arch:         linux/amd64

```
* QEMU:
```
$ rpm -qa | grep qemu
qemu-system-x86-core-2.9.1-1.fc26.x86_64
qemu-block-rbd-2.9.1-1.fc26.x86_64
qemu-block-gluster-2.9.1-1.fc26.x86_64
qemu-common-2.9.1-1.fc26.x86_64
qemu-system-x86-2.9.1-1.fc26.x86_64
qemu-kvm-2.9.1-1.fc26.x86_64
qemu-img-2.9.1-1.fc26.x86_64
libvirt-daemon-driver-qemu-3.2.1-6.fc26.x86_64
ipxe-roms-qemu-20161108-2.gitb991c67.fc26.noarch
qemu-block-nfs-2.9.1-1.fc26.x86_64
qemu-block-dmg-2.9.1-1.fc26.x86_64
qemu-block-ssh-2.9.1-1.fc26.x86_64
qemu-block-iscsi-2.9.1-1.fc26.x86_64
qemu-block-curl-2.9.1-1.fc26.x86_64
qemu-guest-agent-2.9.1-1.fc26.x86_64

```

Start minishift:
```
$ minishift start
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present and active ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'kvm' hypervisor ...
-- Starting Minishift VM .................... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 82% used !!! OK
-- OpenShift cluster will be configured with ...
   Version: v3.6.0
-- Checking `oc` support for startup flags ... 
   host-volumes-dir ... OK
   routing-suffix ... OK
   host-config-dir ... OK
   host-data-dir ... OK
   host-pv-dir ... OK
Starting OpenShift using openshift/origin:v3.6.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.166:8443
```

Stop minishift:
```
$ minishift stop
Stopping local OpenShift cluster...
Cluster stopped.
$ minishift status
Minishift:  Stopped
Profile:    minishift
OpenShift:  Stopped
DiskUsage:  Unknown

```

Start again minishit:
```
$ minishift start
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present and active ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'kvm' hypervisor ...
-- Starting Minishift VM .................... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 82% used !!! OK
-- OpenShift cluster will be configured with ...
   Version: v3.6.0
-- Checking `oc` support for startup flags ... 
   host-config-dir ... OK
   host-data-dir ... OK
   host-pv-dir ... OK
   host-volumes-dir ... OK
   routing-suffix ... OK
Starting OpenShift using openshift/origin:v3.6.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.166:8443

```

HTH

**Comment by javilinux** [comment]
This morning when I tried to reproduced the issue, I couldn't. 

I tried 20+ times in a loop with not a single error.

I will keep the issue open for some days in case I can reproduce it again, and if not, I will close it.

**Comment by gbraad** [comment]
We have observed this behaviour in CI before, but the reasons were unclear. It felt like a timing issue in which the docker mount failed (or storage setup started) and docker started with the wrong location.

**Comment by gbraad** [comment]
https://github.com/minishift/minishift-centos-iso/issues/121

**Comment by gbraad** [comment]
Can you also do a `cat /etc/os-release` inside the Minishift VM? And `sudo journalctl -u minishift-handle-user-data`. Or does a script exist with the name `/use/local/bin/minishift-handle-user-data`?

I rewrote the startup for the CentOS image to cope with this better... Want to make sure this image is used.

**Comment by javilinux** [comment]
Ok, I think I found how to reproduce the issue:
After starting minishift, do a docker pull, then the issue appears:

minishift start
minishift ssh docker pull registry.access.redhat.com/rhscl/nodejs-6-rhel7

If instead of a direct docker pull, I use "oc new-app https://github.com/openshift/nodejs-ex -l name=myapp" the image is pushed to the internal registry and I can stop&start minishift with no issue.

So somehow a direct docker pull and the minishift stop process corrupt /var/lib/docker

Attaching the info requested:
[docker@minishift ~]$ cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

VARIANT="minishift"
VARIANT_VERSION="1.3.0"
BUILD_ID="f29e875-09102017130304-local"


[docker@minishift ~]$ sudo journalctl -u minishift-handle-user-data
-- Logs begin at Wed 2017-11-01 16:44:54 EDT, end at Wed 2017-11-01 16:51:16 EDT. --
Nov 01 16:44:58 localhost systemd[1]: Starting Minishift handle user data...
Nov 01 16:44:58 localhost minishift-handle-user-data[704]: mount p:sda1 ...
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: activate swap ...
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: umount: /var/lib/docker: not mounted
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: total 28
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: drwx------. 2 root root 16384 Oct 26 13:52 lost+found
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: drwxr-xr-x. 2 root root  4096 Oct 26 13:52 opt
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: drwxrwxrwt. 7 root root  4096 Nov  1 16:44 tmp
Nov 01 16:44:59 localhost minishift-handle-user-data[704]: drwxr-xr-x. 3 root root  4096 Oct 26 13:52 var
Nov 01 16:44:59 localhost systemd[1]: Started Minishift hand

[docker@minishift ~]$ ls -l /usr/local/bin/minishift-handle-user-data 
-rwxr-xr-x. 1 root root 5285 Oct  9 08:05 /usr/local/bin/minishift-handle-user-data
le user data.

**Comment by gbraad** [comment]
It seems the handle-user-data is working as expected... can you do a:

```
$ mount
$ ls -al /var/lib/docker
$ sudo journalctl -u docker
```

to debug?

**Comment by gbraad** [comment]
Unable to reproduce. Likely I am missing something...

**Comment by javilinux** [comment]
Sorry for the delay, this is the output:
[docker@minishift ~]$ mount
sysfs on /sys type sysfs (rw,relatime,seclabel)
proc on /proc type proc (rw,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,seclabel,size=1000068k,nr_inodes=250017,mode=755)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,relatime,seclabel)
devpts on /dev/pts type devpts (rw,relatime,seclabel,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,seclabel,mode=755)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,seclabel,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_prio,net_cls)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpuacct,cpu)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
configfs on /sys/kernel/config type configfs (rw,relatime)
/dev/sr0 on /run/initramfs/live type iso9660 (ro,relatime)
/dev/mapper/live-rw on / type ext4 (rw,noatime,seclabel,data=ordered)
rpc_pipefs on /var/lib/nfs/rpc_pipefs type rpc_pipefs (rw,relatime)
selinuxfs on /sys/fs/selinux type selinuxfs (rw,relatime)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=12590)
mqueue on /dev/mqueue type mqueue (rw,relatime,seclabel)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,seclabel)
nfsd on /proc/fs/nfsd type nfsd (rw,relatime)
/dev/sda1 on /mnt/sda1 type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda1 on /var/lib/docker type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda1 on /var/lib/boot2docker type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda1 on /etc/docker type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda1 on /var/lib/minishift type ext4 (rw,relatime,seclabel,data=ordered)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,seclabel,size=204824k,mode=700,uid=1000,gid=1000)
[docker@minishift ~]$ ls -al /var/lib/docker/
ls: cannot open directory /var/lib/docker/: Permission denied
[docker@minishift ~]$ sudo ls -al /var/lib/docker/
total 40
drwx--x--x. 10 root root 4096 Nov  4 06:04 .
drwxr-xr-x. 28 root root 4096 Nov  4 06:13 ..
drwx------.  9 root root 4096 Nov  4 06:08 containers
drwx------.  5 root root 4096 Nov  4 06:05 devicemapper
drwx------.  3 root root 4096 Nov  4 06:04 image
drwxr-x---.  3 root root 4096 Nov  4 06:04 network
drwx------.  2 root root 4096 Nov  4 06:04 swarm
drwx------.  2 root root 4096 Nov  4 06:12 tmp
drwx------.  2 root root 4096 Nov  4 06:04 trust
drwx------.  2 root root 4096 Nov  4 06:04 volumes

Nov 04 06:13:31 localhost systemd[1]: Starting Docker Application Container Engine...
Nov 04 06:13:32 localhost.localdomain dockerd-current[769]: time="2017-11-04T06:13:32.714572225-04:00" level=info msg="libcontainerd: new containerd process, pid: 831"
Nov 04 06:13:33 localhost.localdomain dockerd-current[769]: time="2017-11-04T06:13:33.960428050-04:00" level=warning msg="devmapper: Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man docker` to refer to dm.thinpooldev section."
Nov 04 06:13:33 localhost.localdomain dockerd-current[769]: time="2017-11-04T06:13:33.984447561-04:00" level=warning msg="devmapper: Base device already exists and has filesystem xfs on it. User specified filesystem  will be ignored."
Nov 04 06:13:34 localhost.localdomain dockerd-current[769]: time="2017-11-04T06:13:34.109104552-04:00" level=fatal msg="Error starting daemon: layer does not exist"
Nov 04 06:13:34 localhost.localdomain systemd[1]: docker.service: main process exited, code=exited, status=1/FAILURE
Nov 04 06:13:34 localhost.localdomain systemd[1]: Failed to start Docker Application Container Engine.
Nov 04 06:13:34 localhost.localdomain systemd[1]: Unit docker.service entered failed state.
Nov 04 06:13:34 localhost.localdomain systemd[1]: docker.service failed.

**Comment by gbraad** [comment]
This is consistently reproducible?

You:
  * start, till OpenShift is up.
  * stop
  * start, again, till OpenShift is up
  * followed by
     * docker pull, will fail
     * oc new-app, however succeeds


Right? If so, I can't reproduce this...

**Comment by javilinux** [comment]
This is how I can reproduce it, around 50% of the time:

for i in `seq 10` ; do echo "Try number $1 " ; minishift delete --force ; minishift start --iso-url centos ; minishift ssh docker pull registry.access.redhat.com/rhscl/nodejs-6-rhel7 ; minishift stop ; minishift start ; done

**Comment by gbraad** [comment]
@javilinux Running this now... will check in the morning what the result is.

**Comment by gbraad** [comment]
Have run this now about 3 times, and so far, each pull and restart worked as expected.

Can you describe more about the host it runs on? Is it memory constraint, or are there any other background processes?

**Comment by gbraad** [comment]
Something prevented to start the first time, the pull succeeded, it stopped, and after this it only said "Starting OpenShift". The origin container is running, but the process never continued. From the VM I can see these errors in the docker logs: 

```
Nov 06 22:48:42 new4 oci-systemd-hook[1899]: systemdhook <debug>: Skipping as container command is /bin/bash, not init or systemd
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: prestart /var/lib/docker/devicemapper/mnt/0e677af1ddb4c957492b4fd25799e8e7b9a51793d11c46bcca243e5aee5fe7ca/rootfs
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker/overlay2]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker/overlay]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker-latest/overlay2]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker-latest/overlay]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker-latest/devicemapper]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/docker-latest/containers/]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/container/storage/lvm]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/container/storage/devicemapper]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Failed to canonicalize path [/var/lib/container/storage/overlay]: No such file or directory. Skipping.
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: mapped host_mnt=/var/lib/docker/devicemapper to cont_mnt=/rootfs/var/lib/docker/devicemapper
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: Unmounted: [/var/lib/docker/devicemapper/mnt/0e677af1ddb4c957492b4fd25799e8e7b9a51793d11c46bcca243e5aee5fe7ca/rootfs/rootfs/var/lib/docker/devicemapper]
Nov 06 22:48:42 new4 oci-umount[1900]: umounthook <info>: mapped host_mnt=/var/lib/docker/containers to cont_mnt=/rootfs/var/lib/docker/containers
```

**Comment by gbraad** [comment]
@LalatenduMohanty @praveenkumar 
Can we try to switch to overlay ?

**Comment by LalatenduMohanty** [comment]
> Can we try to switch to overlay ?

+1. Minikube ISO uses overlay so can we try it with the latest master build http://artifacts.ci.centos.org/minishift/minishift/master/latest/

```
minishift start --iso-url minikube
```

**Comment by LalatenduMohanty** [comment]
Resolved by #1961
